examples/converters.py

```diff
@@ -0,0 +1,115 @@
+# This example requires the 'members' privileged intent to use the Member converter.
+
+import typing
+
+import discord
+from discord.ext import commands
+
+intents = discord.Intents.default()
+intents.members = True
+
+bot = commands.Bot('!', intents=intents)
+
+
+@bot.command()
+async def userinfo(ctx: commands.Context, user: discord.User):
+    # In the command signature above, you can see that the `user`
+    # parameter is typehinted to `discord.User`. This means that
+    # during command invocation we will attempt to convert
+    # the value passed as `user` to a `discord.User` instance.
+    # The documentation notes what can be converted, in the case of `discord.User`
+    # you pass an ID, mention or username (discrim optional)
+    # E.g. 80088516616269824, @Danny or Danny#0007
+
+    # NOTE: typehinting acts as a converter within the `commands` framework only.
+    # In standard Python, it is use for documentation and IDE assistance purposes.
+
+    # If the conversion is successful, we will have a `discord.User` instance
+    # and can do the following:
+    user_id = user.id
+    username = user.name
+    avatar = user.avatar_url
+    await ctx.send('User found: {} -- {}\n{}'.format(user_id, username, avatar))
+
+@userinfo.error
+async def userinfo_error(ctx: commands.Context, error: commands.CommandError):
+    # if the conversion above fails for any reason, it will raise `commands.BadArgument`
+    # so we handle this in this error handler:
+    if isinstance(error, commands.BadArgument):
+        return await ctx.send('Couldn\'t find that user.')
+
+# Custom Converter here
+class ChannelOrMemberConverter(commands.Converter):
+    async def convert(self, ctx: commands.Context, argument: str):
+        # In this example we have made a custom converter.
+        # Specifically - to check if an input is convertable to a
+        # `discord.Member` or `discord.TextChannel` instance from a
+        # single input.
+        # E.g. we have an ID we want to add to a list of ignored IDs, or similar
+
+        member_converter = commands.MemberConverter()
+        try:
+            # Try and convert to a Member instance.
+            member = await member_converter.convert(ctx, argument)
+        except commands.MemberNotFound:
+            # Could not convert to a Member instance.
+            pass
+        else:
+            # We have our `member` so lets return here.
+            return member
```

Comment:
```suggestion
        try:
            # Try and convert to a Member instance.
            # If this fails, then an exception is raised.
            # Otherwise, we just return the converted member value.
            member = await member_converter.convert(ctx, argument)
        except commands.MemberNotFound:
            pass
        else:
            return member
```